src/odb/src/db/dbBTerm.cpp

```diff
@@ -927,19 +927,17 @@ void dbBTerm::staSetVertexId(uint32_t id)
   iterm->_sta_vertex_id = id;
 }
 
-void dbBTerm::setConstraintRegion(
-    const std::pair<Point, Point>& constraint_region)
+void dbBTerm::setConstraintRegion(const Rect& constraint_region)
 {
   _dbBTerm* bterm = (_dbBTerm*) this;
   bterm->_constraint_region = constraint_region;
 }
 
-std::optional<std::pair<Point, Point>> dbBTerm::getConstraintRegion()
+std::optional<Rect> dbBTerm::getConstraintRegion()
 {
   _dbBTerm* bterm = (_dbBTerm*) this;
   const auto& constraint_region = bterm->_constraint_region;
-  if (constraint_region.first == Point(0, 0)
-      && constraint_region.second == Point(0, 0)) {
+  if (constraint_region == Rect(0, 0, 0, 0)) {
```

Comment:
While we are improving this, it would be better to use an inverted rect rather than the origin as the no value - see mergeInit & isInverted